Providers/Scripts/2.6x-2.7x/Scripts/nxAvailableUpdates.py

```diff
@@ -125,44 +129,89 @@ def GetYumUpdates(Name):
             yum_info = 'repoquery --queryformat "Name : %{NAME}\nArch : %{ARCH}\nVersion :  %{EPOCH}:%{VERSION}\nRelease : %{RELEASE}\nRepo : %{REPO}\nBuildtime : %{BUILDTIME}" '
         else:
             yum_info = 'yum info available '
-    cmd = "LANG=en_US.UTF8 " + yum_list + "| awk '{print $1}'"
-    code, pkg_list = RunGetOutput(cmd, False, False)
-    if len(pkg_list) < 2 :
-        LG().Log('INFO', "Nothing to send info on. No packages")
-        return updates_list
-    # Remove the chatter.  Yum puts a blank line before the list.
-    if pkg_list.find('\n\n') > -1:
-        pkg_list = pkg_list[pkg_list.find('\n\n')+2:]
-    LG().Log('DEBUG', "Number of packages: " + str(len(pkg_list.splitlines())))
-    srch = re.compile(srch_str, re.M | re.S)
-    for pkg in pkg_list.splitlines():
-        if len(pkg) < 2 :
-            LG().Log('VERBOSE', "Avoiding very small entries.")
-            continue
-        cmd = "LANG=en_US.UTF8 " + yum_info + pkg
+    cmd = "LANG=en_US.UTF8 " + yum_list
+    start_time = time.time()
+    LG().Log('DEBUG', "Retrieving update package list using cmd: " + cmd)
+    retcode, pkg_list = RunGetOutput(cmd, False, False)
+    LG().Log('DEBUG', "Cmd execution time: " + str((time.time() - start_time) / 60))
+    #LG().Log('DEBUG', "Cmd output for update list : " + pkg_list)    
+
+    # Sample output from OMSYumUpdates.sh
+    #   ca-certificates.noarch
+    #   glibc.x86_64
+    #   glibc-common.x86_64
+    #   glibc-devel.x86_64
+    #   glibc-headers.x86_64
+    if validate_yum_pkg_list_output(pkg_list, retcode):
+        # Remove the chatter.  Yum puts a blank line before the list.
+        if pkg_list.find('\n\n') > -1:
+            pkg_list = pkg_list[pkg_list.find('\n\n') + 2:]
+        LG().Log('DEBUG', "Number of packages to be updated: " + str(len(pkg_list.splitlines())))
+        srch = re.compile(srch_str, re.M | re.S)
+
+        param_list = ""
+        for pkg in pkg_list.splitlines():
+            param_list = param_list + " " + pkg
+
+        cmd = "LANG=en_US.UTF8 " + yum_info + param_list
+        LG().Log('DEBUG', "Retrieving individual package information from Yum using cmd: " + cmd)
+        start_time = time.time()
         code, out = RunGetOutput(cmd, False, False)
+        LG().Log('DEBUG', "Cmd execution time: " + str((time.time() - start_time) / 60))
+        #LG().Log('DEBUG', "Cmd output is : " + out)
+        updates_list = []
         if len(out) < 1 or ':' not in out:
-            LG().Log('DEBUG', "Checking the output in 'out': " + out)
-            continue
-        m = srch.search(out)
-        if m == None:
-            continue
-        d['Name'] = m.group(1)
+            LG().Log("Failed retrieving individual package info. Output is too small : " + out)
```

Comment:
Same issue as before with not using the log level parameter.
Don't forget to propagate all fixes into the 3 Python files!